### PR TITLE
Csweaver/rest diffexp

### PR DESCRIPTION
**Pull request by csweaver**
added t-test diffexp for scanpy engine. This is the quick and dirty version of /diffexp/obs. Anything not absolutely necessary for getting the FE running has been ignored.

a little unsure about Bonferroni correction (@freeman-lab can you please check that the stats make sense?) Bonferroni not at right sigfig currently for sure.
@ttung wasn't sure if the _top_sort method should be in the engine or utils. Went with engine since I didn't imagine using it elsewhere

### CODE REVIEW

File: server/test/test_api.py [hunk 1]
```diff
@@ -147,6 +147,34 @@ def test_filter_put_annotations_obs(self):
         self.assertEqual(len(result_data["data"][0]), 3)
         self.assertEqual(len(result_data["data"]), 15)
 
+    def test_diff_exp(self):
+        endpoint = "diffexp/obs"
+        url = f"{URL_BASE}{endpoint}"
+        params = {
```
Comment (bkmartinjr) [inline server/test/test_api.py]: This test does not conform to the spec.    mode should be an object and contain "count" and "name", eg,

{  mode:  { count: 7, name: "topN" } }

File: server/test/test_api.py [hunk 2]
```diff
@@ -147,6 +147,34 @@ def test_filter_put_annotations_obs(self):
...
+            "mode": "topN",
+            "set1": {
+                "filter": {
+                    "obs": {"annotation_value": [
+                        {"name": "louvain", "values": ["NK cells"]}
+                    ]
+                    }
+                }
+            },
+            "set2": {
```
Comment (bkmartinjr) [inline server/test/test_api.py]: super minor point - the current f/e usage is filtering by index, not annotation value.   Might want to add a test for that?

File: server/test/test_api.py (hunk 1 above)
Comment (bkmartinjr) [reply server/test/test_api.py]: spec is wrong (inconsistent) - will fix it.

File: server/test/test_api.py (hunk 2 above)
Comment (csweaver) [reply server/test/test_api.py]: got it

**Pull request merged**